dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/fat/src/io/openliberty/webcontainer/servlet61/fat/tests/Servlet61ReadWriteByteBufferTest.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.webcontainer.servlet61.fat.tests;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Test ServletInputStream.read(ByteBuffer) and ServletOutputStream.write(ByteBuffer)
+ */
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class Servlet61ReadWriteByteBufferTest {
+    private static final Logger LOG = Logger.getLogger(Servlet61ReadWriteByteBufferTest.class.getName());
+    private static final String TEST_APP_NAME = "ReadWriteByteBuffer";
+    private static final String POST_DATA = "ABCDEFGHIJKLMNOPQRSTUVWXZY_1234567890@0987654321";
+
+    @Server("servlet61_ReadWriteByteBuffer")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        LOG.info("Setup : servlet61_ReadWriteByteBuffer.");
+        ShrinkHelper.defaultDropinApp(server, TEST_APP_NAME + ".war", "readwritebytebuffer.servlets");
+
+        ArrayList<String> expectedErrors = new ArrayList<String>();
+        expectedErrors.add("CWWWC0009E:.*"); //Write Null ByteBuffer
```

Comment:
Is this correct? We're adding `CWWWC0010E` and `CWWWC0011E` in the product code.